jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/archive/AssemblyFileSetUtilsProcessAssemblyFileSetTest.java

```diff
@@ -0,0 +1,343 @@
+        .build();
+    final AssemblyConfiguration ac = AssemblyConfiguration.builder()
+        .name("deployments")
+        .build();
+    // When
+    final Map<File, String> permissions = AssemblyFileSetUtils.processAssemblyFileSet(baseDirectory, outputDirectory, afs, ac);
+    // Then
+    assertThat(permissions.entrySet(), hasSize(4));
+    final File deployments = new File(outputDirectory, "deployments");
+    assertThat(deployments.exists(), equalTo(true));
+    assertThat(deployments.listFiles(), arrayWithSize(1));
+    final File outputSourceDir = Objects.requireNonNull(deployments.listFiles())[0];
+    assertThat(outputSourceDir.getName(), equalTo("source-directory"));
+    assertThat(outputSourceDir.exists(), equalTo(true));
+    assertThat(outputSourceDir.listFiles(), arrayWithSize(6));
+    assertThat(outputSourceDir.list(), arrayContainingInAnyOrder("one", "two", "three", "1.txt", "3.other", "37"));
+  }
+
+  /**
+   * Has AssemblyFileSet#directory and AssemblyConfiguration#name options.
+   *
+   * Source directory doesn't exist
+   *
+   * Should do nothing.
+   */
+  @Test
+  public void sourceDoesNotExist() throws Exception {
+    // Given
+    final AssemblyFileSet afs = AssemblyFileSet.builder()
+        .directory(new File(sourceDirectory, "non-existent"))
+        .build();
+    final AssemblyConfiguration ac = AssemblyConfiguration.builder()
+        .name("deployments")
+        .build();
+    // When
+    final Map<File, String> permissions = AssemblyFileSetUtils.processAssemblyFileSet(baseDirectory, outputDirectory, afs, ac);
+    // Then
+    assertThat(permissions.entrySet(), empty());
+    final File deployments = new File(outputDirectory, "deployments");
+    assertThat(deployments.exists(), equalTo(false));
+  }
+
+  /**
+   * Has AssemblyFileSet with directory and outputDirectory relative path resolving to self.
+   * Has AssemblyConfiguration name.
+   *
+   * Should copy contents of AssemblyFileSet#directory to the outputDirectory .
+   */
+  @Test
+  public void fileSetDirectoryAndOutputDirectoryResolvingToSelf() throws Exception {
+    // Given
+    final AssemblyFileSet afs = AssemblyFileSet.builder()
+        .directory(sourceDirectory)
+        .outputDirectory(new File("."))
+        .build();
+    final AssemblyConfiguration ac = AssemblyConfiguration.builder()
+        .name("deployments")
+        .build();
+    // When
+    final Map<File, String> permissions = AssemblyFileSetUtils.processAssemblyFileSet(baseDirectory, outputDirectory, afs, ac);
+    // Then
+    assertThat(permissions.entrySet(), hasSize(4));
+    final File deployments = new File(outputDirectory, "deployments");
+    assertThat(deployments.exists(), equalTo(true));
+    assertThat(deployments.listFiles(), arrayWithSize(6));
+    assertThat(deployments.list(), arrayContainingInAnyOrder("one", "two", "three", "1.txt", "3.other", "37"));
+    final File outputSourceDir = Objects.requireNonNull(deployments.listFiles())[0];
```

Comment:
`outputSourceDir` seems unused. Umm, maybe you forgot to assertions for it? If not, could you please remove it?

Reply:
Yep, some descriptions are wrong too. I changed the tests when I figured out some scenarios were wrongly defined.